### DISCUSSION TITLE
How to listen event triggered by clicking the app icon on the dock(macos) or taskbar(windows)

### DISCUSSION THREAD

**Discussion by lisiur** [tauri-apps/tauri#6600] [Q&A] [answered]
Say I hide the main window, how can I show the window when user click the icon in the dock or taskbar.

**Comment by FabianLars** [comment, marked as answer]
This is not possible yet, it's tracked here: https://github.com/tauri-apps/tauri/issues/3084

**Comment by billmorgan123** [comment]
you can do it on app.on('activate')
`app.on('activate', () => {
  if (mainWindow === null) {
    createWindow()
  } else {
    if(!mainWindow.isVisible()) {
      mainWindow.show()
    }
  }
})`

**Reply by FabianLars** [reply]
This is electron specific code that does not work in tauri.

**Comment by gustav87** [comment]
Hi!

Is this also fixed for Windows? After our app has been minimized to taskbar, or minimized to system tray, we would like it to show if you click the icon on the desktop.

Many of our users think our app is fully closed when clicking the X, but it's actually minimized to the system tray. The user then clicks the app icon on their desktop to start it again, but nothing happens, because it's already started. It does not show if it's minimized and the user clicks on the desktop icon. You need to navigate to the system tray to find it and click it there.

**Reply by FabianLars** [reply]
Do you mean the desktop icon on well, the desktop? (Asking because this discussion is specifically about the taskbar)

That should always open a new app instance whether or not it's running normally or in the tray. Are you using the single-instance plugin perhaps? That should trigger then and make it look like there's nothing happening.

I'm not sure about the taskbar icon, i would have expected it to open a new instance when there's no window associated with the taskbar icon but i never tested it. There's also no explicit event like on macos in tauri, not sure what kind of apis/events the OS has to offer here for us.

**Reply by gustav87** [reply]
Yes, the desktop icon on the desktop. Yes, indeed we are using the single-instance plugin. While the app is minimized to tray, some users doesn't realize the app is running, so they click on the desktop icon, and nothing happens. They think something is wrong, and reach out to us for help.

Is there some way the app could "pop up" when clicked on the desktop?

**Reply by gustav87** [reply]
Actually, I see the answer on the single-instance plugin page: https://v2.tauri.app/plugin/single-instance/#focusing-on-new-instance

Thank you for your answer and pointing me in the right direction!

**Reply by phattranky** [reply]
I see this solution works for me https://github.com/tauri-apps/tauri/issues/3084#issuecomment-1477675840
